### PR TITLE
#165952409 Implement log in endpoint

### PR DESCRIPTION
**Pull request by Sojisoyoye**
**What does this PR do?**
 
implemented `/auth/login` endpoint

**Description of Task to be completed?**

create `/auth/login` route
validates input email
check if user email exist
compares the password with the encrypted database password
sends a JSON Obj response with token

**How should this be manually tested?**

Clone this branch
Run `npm run start-dev`
Send a post request to http://127.0.0.1:3000/api/v1/auth/login

**Any background context you want to provide?**

Post data model { `email:` string, email , `password:` string }

### CODE REVIEW

**Comment by Sojisoyoye** [comment]
@frankly034, @Eazybee, @EmmanuelSage , @mosinmiloluwa01 , kindly review.

**Review by EmmanuelSage — COMMENTED** [review]
This is a good one soji,
I have a question about the sequelize cli in package.json,
it seems to have been deleted?

File: package.json [hunk 1]
```diff
@@ -51,7 +51,6 @@
     "eslint-plugin-jsx-a11y": "^6.2.1",
     "eslint-plugin-react": "^7.13.0",
     "mocha": "^6.1.4",
-    "sequelize-cli": "^5.4.0",
```
Comment (Eazybee) [inline package.json:54]: it seems you must have mistakenly deleted sequelize-cli form your package.json

File: src/db/config/config.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "development": {
-    "username": "postgres",
-    "password": "onelove1!",
+    "username": "olusojisoyoye",
```
Comment (Eazybee) [inline src/db/config/config.json]: please endeavor to  always change username and password to `postgres` before pushing it

File: src/server/middlewares/validator.js [hunk 3]
```diff
@@ -32,5 +32,29 @@ class Validate {
       DisplayError(err, res);
     }
   }
+
+  static login(req, res, next) {
+    const { email } = req.body;
+
+    const data = {
+      email
```
Comment (Eazybee) [inline src/server/middlewares/validator.js]: you might want to validate password as required

File: src/server/controllers/auth.js [hunk 4]
```diff
@@ -39,6 +39,38 @@ class auth {
       })
       .catch(e => res.status(400).send({ status: 400, error: `Bad request ${e}` }));
   }
+
+  static login(req, res) {
+    const { email, password } = req.body;
+
```
Comment (Eazybee) [inline src/server/controllers/auth.js:48]: you might want  to convert email to lower case before searching db

**Comment by Sojisoyoye** [comment]
Yes, I deleted it, because there was sequelize-cli in both dependencies and dev-dependencies, which is not supposed to be, so I had to delete the one in dev-dependencies.

**Comment by EmmanuelSage** [comment]
Oh okay, it is fine.
Good one bro.

**Comment by Sojisoyoye** [comment]
@Eazybee, kindly help verify if all the feedbacks as being correctly implemented

**Review by Eazybee — CHANGES_REQUESTED** [review]

**Comment by Sojisoyoye** [comment]
@Eazybee, kindly review

**Review by Eazybee — APPROVED** [review]

**Pull request merged**